lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -882,6 +878,29 @@ def install_puppetlabs_release_repo_on( hosts, repo = nil, opts = options )
         end
         alias_method :install_puppetlabs_release_repo, :install_puppetlabs_release_repo_on
 
+        # Installs the repo configs on a given host
+        #
+        # @param [Beaker::Host] host Host to install configs on
+        # @param [String] buildserver_url URL of the buildserver
+        # @param [String] package_name Name of the package
+        # @param [String] build_version Version of the package
+        # @param [String] copy_dir Local directory to fetch files into & SCP out of
+        #
+        # @return nil
+        def install_repo_configs(host, buildserver_url, package_name, build_version, copy_dir)
+          repo_filename = host.repo_filename( package_name, build_version )
+          repo_config_folder_url = "%s/%s/%s/repo_configs/%s/" %
+            [ buildserver_url, package_name, build_version, host.repo_type ]
+
+          repo = fetch_http_file( repo_config_folder_url,
+                                  repo_filename,
+                                  copy_dir )
+          scp_to( host, repo, host.package_config_dir )
+
+          on( host, 'apt-get update' ) if host['platform'] =~ /ubuntu-15\.?10/
```

Comment:
The issue is that our ubuntu < 15.10 VMs already have ntpdate installed, so when we go to install `puppet-agent`, `Host::Unix::Pkg#apt_needs_update` is still true, so we do `apt-get update` before installing `puppet-agent`. Our ubuntu 15.10 VM does not have ntpdate installed, so we do `apt-get update` prior to installing `ntpdate` early in the beaker run. Then we install the puppet-agent repo config, but by now `apt_needs_update` is false, so we don't `apt-get update` again. I think we should just always run `apt-get update` after installing the puppet_agent_dev_repo like we used to in https://github.com/puppetlabs/beaker/pull/1044/files#diff-b0b73189db06b67008309aabcc69a743L1020.

Sorry for leading us down this path. I didn't understand why apt-get update was running multiple times earlier. I'll also follow up with RE about why the ubuntu VMs are not consistent.